templates/_schemas/definitions/webpage.twig

```diff
@@ -1,21 +1,13 @@
 {% macro webpage(props, schema = null) %}
-  {% from '_schemas/definitions/thing' import thing %}
+  {% from '_schemas/definitions/creativeWork' import creativeWork %}
   {% from '_schemas/definitions/imageObject' import imageObject %}
-  {% from '_schemas/definitions/organization' import organization %}
   
-  {% set schema = schema ?? craft.schema.Webpage %}
-  {% set schema = thing(props, schema) %}
+  {% set schema = schema ?? craft.schema.WebPage %}
+  {% set schema = creativeWork(props, schema) %}
   {% set schema = schema
-    .id(props.url ~ '#webpage')
-    .dateCreated(props.dateCreated)
-    .datePublished(props.dateCreated)
-    .dateModified(props.dateUpdated)
-    .headline(create('craft\\helpers\\StringHelper').safeTruncate(props.title, 110))
-    .primaryImageOfPage((props.images ?? []) is not empty ? imageObject(props.images|first) : null)
-    .author((props.authors.organizations ?? [])|map(org => organization(org))|default(null))
-    .creator((props.creators ?? [])|map(org => organization(org))|default(null))
-    .copyrightHolder((props.authors.organizations ?? [])|map(org => organization(org))|default(null))
-    .copyrightYear(props.dateCreated|date('Y'))
+    .id(siteUrl|default(null))
```

Comment:
Is this correct? All webpage `id` values will end up being the same. Previously this was being set to `props.url ~ '#webpage'`.

Reply:
On [reference URL](https://github.com/miranj/miranj.in/commit/98cd544ec95c6863eb84c18b4ef63a39d47689dc), it was updated intentionally to `siteUrl` so I did here as well but I think keeping it `props.url ~ '#webpage'` would be better. Reverted now.

Reply:
What is the reason for one being better than the other?

Reply:
Keeping it `siteUrl` will make it the same for all web pages and the `Id` should be unique for each webpage.

Reply:
@pl-mnm Can you chime in here? The older code in the boilerplate has the webpage ID set to `props.url ~ '#webpage'` but the newer versions of our JSON-LD code, like in Blume, have the ID set to `siteUrl`.

https://github.com/miranj/craft-boilerplate/blob/531a643f16af781443be448e7d05ca5576136f68/templates/_schemas/definitions/webpage.twig#L9

https://github.com/miranj/blume-fresh/blob/6dfa86f17ec475f8e033cb3a299a8caa1929f664/templates/_schemas/definitions/webpage.twig#L10

- What is the ID supposed to be?
- What should we retain in the boilerplate?

Reply:
Tbh I'm not sure I understood all these subtleties. This is something @souvikdg and I had come up with and I think he'll be better placed to tell us what's required.